### PR TITLE
[main] check dataset and interval size during inference

### PR DESCRIPTION
**Pull request by tijyojwad**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -250,6 +250,12 @@ def infer_worker(gpu, ngpu_per_node, args, res_queue=None):
     infer_dataset = DatasetInfer(args.infer_files)
     infer_sampler = None
 
+    # Check that the intervals file and infer data have same number of entries.
+    intervals = pd.read_csv(args.intervals_file, sep='\t', header=None, names=['chrom', 'start', 'end'],
```
Comment (avantikalal) [inline main.py]: Why do we need to distribute this check to every GPU? Would it be better placed in `main()`?

Comment (tijyojwad) [reply main.py]: Good point. Moved it over

File: main.py [hunk 2]
```diff
@@ -529,6 +529,15 @@ def main():
                 args.interval_size = f['data'].shape[1]
                 args.batch_size = 1
 
+            # Ensure the sizes for intervals file and infer dataset are same.
+            infer_dataset = DatasetInfer(args.infer_files)
```
Comment (avantikalal) [inline main.py]: This should only be run if `args.infer` is true, not `args.eval`.

File: main.py (hunk 1 above)
Comment (tijyojwad) [reply main.py]: done

File: main.py (hunk 2 above)
Comment (tijyojwad) [reply main.py]: done

**Pull request merged**